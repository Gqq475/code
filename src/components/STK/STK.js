import React from 'react'
import classes from './STK.scss'

type Props = {

};
export class STK extends React.Component {
  props: Props;

  constructor (props) {
    super(props)
    this.state = {
      url: ''
    }
  }

  componentWillMount () {
    let url = 'http://172.22.35.131:8080/workbooks/' +
      '74596612-9df5-42aa-b4ed-759c9213d184/views/296f2b3d-bf88-444f-8e9f-4775d3e3b2e7'
    fetch(url)
      .then((res) => {
        return res.json()
      })
      .then((data) => {
        this.setState({ url: data.result.contenturl })
      })
  }

  render () {
    return (
      <div className={classes['STK-container']}>
        {
          <iframe className={classes.iframe} src={this.state.url} />
        }
      </div>
    )
  }
}

export default STK
