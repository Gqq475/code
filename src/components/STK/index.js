import STK from './STK'
export default STK
