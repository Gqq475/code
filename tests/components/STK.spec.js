import React from 'react'
import STK from 'components/STK/STK'

describe('(Component) STK', () => {
  it('render normally', () => {
    const wrapper = shallow(
      <STK />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
