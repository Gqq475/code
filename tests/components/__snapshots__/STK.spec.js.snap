// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) STK render normally 1`] = `
<div
  className="STK-container"
>
  <iframe
    className="iframe"
    src=""
  />
</div>
`;
